mm2src/mm2_main/src/lp_swap/swap_events.rs

```diff
@@ -0,0 +1,51 @@
+use super::maker_swap::MakerSavedEvent;
+use super::maker_swap_v2::MakerSwapEvent;
+use super::taker_swap::TakerSavedEvent;
+use super::taker_swap_v2::TakerSwapEvent;
+use mm2_event_stream::{Broadcaster, Event, EventStreamer, StreamHandlerInput};
+
+use async_trait::async_trait;
+use futures::channel::oneshot;
+use futures::StreamExt;
+use uuid::Uuid;
+
+pub struct SwapStatusStreamer;
+
+impl SwapStatusStreamer {
+    pub fn new() -> Self { Self }
+
+    pub fn derive_streamer_id() -> &'static str { "SWAP_STATUS" }
+}
```

Comment:
little patch for optimizing this:

```suggestion
impl SwapStatusStreamer {
    #[inline(always)]
    pub fn new() -> Self { Self }

    #[inline(always)]
    pub const fn derive_streamer_id() -> &'static str { "SWAP_STATUS" }
}
```

Reply:
[547fc2b](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2172/commits/547fc2bb4c03980379cfd219995e1d5ffa199bde)